filecoin-proofs/src/bin/paramfetch.rs

```diff
@@ -81,77 +120,162 @@ fn fetch(matches: &ArgMatches) -> Result<()> {
         )));
     }
 
-    let parameter_map = get_parameter_map(&json_path)?;
-    let all_parameter_ids = get_mapped_parameter_ids(&parameter_map)?;
+    let manifest = read_parameter_map_from_disk(&json_path)?;
+    let mut filenames = get_filenames_from_parameter_map(&manifest)?;
+
+    println!("{} files in manifest...", filenames.len());
+    println!();
+
+    // if user has specified sector sizes for which they wish to download Groth
+    // parameters, trim non-matching Groth parameter filenames from the list
+    if matches.is_present("params-for-sector-sizes") {
+        let whitelisted_sector_sizes: Vec<u64> =
+            values_t!(matches.values_of("params-for-sector-sizes"), u64)?;
+
+        let whitelisted_sector_sizes: HashSet<u64> =
+            HashSet::from_iter(whitelisted_sector_sizes.iter().cloned());
```

Comment:
You should be able to collect into `HashSet` directly, without going through the `Vec` first

Reply:
Done. Good call.